### PR TITLE
Rover FS gcs timeout

### PR DESCRIPTION
**Pull request by khancyr**
Add FS_GCS_TIMEOUT for rover
add gcs text when FS trigger and rover continue AUTO mission
add test for it
add test for it on copter

Slight issue, the implementation on rover differs from copter one's. On rover we already have FS_TIMEOUT that decide when to trigger the failsafe.
Therefor the failsafe is always triggered on max( FS_TIMEOUT, FS_GCS_TIMEOUT), that was already the case before having the parameter.

close https://github.com/ArduPilot/ardupilot/pull/13539

### CODE REVIEW

File: Rover/Parameters.h [hunk 1]
```diff
@@ -433,6 +433,8 @@ class ParametersG2 {
 
     // manual mode steering expo
     AP_Float manual_steering_expo;
+
+    AP_Float fs_gcs_timeout;
```
Comment (rmackay9) [inline Rover/Parameters.h:437]: let's put a little comment above this parameter so it matches the others.  I know that its purpose is pretty clear from the name but..

**Review by rmackay9 — APPROVED** [review]
This looks quite good to me, thanks!

File: Rover/Parameters.h (hunk 1 above)
Comment (khancyr) [reply Rover/Parameters.h:438]: done

File: Rover/Rover.cpp [hunk 2]
```diff
@@ -344,22 +344,23 @@ void Rover::ahrs_update()
...
+    if (g.fs_gcs_enabled == FS_GCS_DISABLED) {
         // gcs failsafe disabled
         return;
     }
 
-    // check for updates from GCS within 2 seconds
     const uint32_t gcs_last_seen_ms = gcs().sysid_myggcs_last_seen_time_ms();
-    bool do_failsafe = true;
     if (gcs_last_seen_ms == 0) {
-        // we've never seen the GCS, so we never failsafe for not seeing it
```
Comment (peterbarker) [inline Rover/Rover.cpp:356]: This comment is quite a useful one and we shouldn't lose it.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Looks good apart from the comment

File: Rover/Rover.cpp (hunk 2 above)
Comment (khancyr) [reply Rover/Rover.cpp:356]: added back

**Comment by khancyr** [comment]
comments addressed, Pr for wiki here : https://github.com/ArduPilot/ardupilot_wiki/pull/4964

**Review by peterbarker — APPROVED** [review]

**Pull request merged**